src/graph.cc

```diff
@@ -31,7 +31,16 @@
 using namespace std;
 
 bool Node::Stat(DiskInterface* disk_interface, string* err) {
-  return (mtime_ = disk_interface->Stat(path_, err)) != -1;
+  METRIC_RECORD("node stat");
+  mtime_ = disk_interface->Stat(path_, err);
+  exists_ = (mtime_ > 0) ? ExistenceStatusExists : ExistenceStatusMissing;
```

Comment:
I think checking for mtime_ > 0 is not consistent with previous behavior. In graph.h the exists() function was implemented by checking mtime_ != 0 and not > 0. This should be fine as long as mtime_ can not have negative values, like -1000. Unfortunately, we observed that on windows for a file dated 31.12.1979 23:00 the mtime return by disk_interface->Stat is a very large negative number leading to ninja complaining about missing files that actually exist.

We will try with the following fix:
```suggestion
  exists_ = (mtime_ != 0) ? ExistenceStatusExists : ExistenceStatusMissing;
```

Nevertheless, thanks for this PR, it saved us a lot of time! We are using it in a build system with hundreds of users.

Reply:
Great catch! Looking at the RealDiskInterface::Stat code, it looks like it adjusts the Windows epoch to the year 2000, which would explain the negative value. Unfortunately, Stat() also returns -1 on error. In that case, we should probably continue to treat the existence status as unknown.

IMO, it's probably less than ideal that mtimes could be negative values, but I believe adjusting the epoch would break existing build logs. That's probably outside the scope of this PR, so I'll match existing behavior for now.